src/gtest/test_libzendoo.cpp

```diff
@@ -2491,3 +2500,409 @@ TEST(CctpLibrary, TestCustomFieldsValidationFix)
+    CMutableTransaction mtx;
+    mtx.nVersion = SC_TX_VERSION;
+    mtx.vsc_ccout.resize(0);
+    mtx.vft_ccout.resize(0);
+    mtx.vmbtr_out.resize(0);
+    mtx.vcsw_ccin.resize(0);
+
+    // forward transfers
+    for (int i = 0; i < CCTP_COMMITMENT_BUILDER_FT_LIMIT; i++)
+        mtx.vft_ccout.push_back(CTxForwardTransferOut(sidechainId, CAmount(43), uint256S("abba101"), uint160S("abba101")));
+
+    CTransaction tx(mtx);
+    ASSERT_TRUE(cmtObj.add(tx));
+
+    CMutableTransaction mtx3;
+    mtx3.nVersion = SC_TX_VERSION;
+    mtx3.vsc_ccout.resize(0);
+    mtx3.vft_ccout.resize(0);
+    mtx3.vmbtr_out.resize(0);
+    mtx3.vcsw_ccin.resize(0);
+
+    mtx3.vft_ccout.push_back(CTxForwardTransferOut(sidechainId, CAmount(43), uint256S("abba101"), uint160S("abba101")));
+
+    CTransaction tx3(mtx3);
+    ASSERT_FALSE(cmtObj.add(tx3));
+
+    // check counters
+    auto cbgRef  = cmtObj.getCBG();
+    ASSERT_EQ(cbgRef.cbgaMap.size(), 1);
+    ASSERT_EQ(cbgRef.cbgcMap.size(), 0);
+    ASSERT_EQ(cbgRef.cbgaMap[sidechainId].ft,   CCTP_COMMITMENT_BUILDER_FT_LIMIT);
+    ASSERT_EQ(cbgRef.cbgaMap[sidechainId].bwtr, 0);
+    ASSERT_EQ(cbgRef.cbgaMap[sidechainId].cert, 0);
+}
+
+TEST(CctpLibrary, CommitmentBuilder_toomanyBWTR)
+{
+    SidechainTxsCommitmentBuilder cmtObj;
+    uint256 sidechainId = uint256S("abc");
+    CMutableTransaction mtx;
+    mtx.nVersion = SC_TX_VERSION;
+    mtx.vsc_ccout.resize(0);
+    mtx.vft_ccout.resize(0);
+    mtx.vmbtr_out.resize(0);
+    mtx.vcsw_ccin.resize(0);
+
+    Sidechain::ScBwtRequestParameters scbwtr_params;
+    scbwtr_params.vScRequestData.push_back(CFieldElement{SAMPLE_FIELD});
+    scbwtr_params.scFee = 3;
+
+    // backward transfer requests
+    for (int i = 0; i < CCTP_COMMITMENT_BUILDER_BWTR_LIMIT; i++)
+        mtx.vmbtr_out.push_back(CBwtRequestOut(sidechainId, uint160S("abba101"), scbwtr_params));
+
+    CTransaction tx(mtx);
+    ASSERT_TRUE(cmtObj.add(tx));
+
+    CMutableTransaction mtx3;
+    mtx3.nVersion = SC_TX_VERSION;
+    mtx3.vsc_ccout.resize(0);
+    mtx3.vft_ccout.resize(0);
+    mtx3.vmbtr_out.resize(0);
+    mtx3.vcsw_ccin.resize(0);
+
+    mtx3.vmbtr_out.push_back(CBwtRequestOut(sidechainId, uint160S("abba101"), scbwtr_params));
+
+    CTransaction tx3(mtx3);
+    ASSERT_FALSE(cmtObj.add(tx3));
+
+    // check counters
+    auto cbgRef  = cmtObj.getCBG();
+    ASSERT_EQ(cbgRef.cbgaMap.size(), 1);
+    ASSERT_EQ(cbgRef.cbgcMap.size(), 0);
+    ASSERT_EQ(cbgRef.cbgaMap[sidechainId].ft,   0);
+    ASSERT_EQ(cbgRef.cbgaMap[sidechainId].bwtr, CCTP_COMMITMENT_BUILDER_BWTR_LIMIT);
+    ASSERT_EQ(cbgRef.cbgaMap[sidechainId].cert, 0);
+}
+
+TEST(CctpLibrary, CommitmentBuilder_toomanyCERT) {
+    SelectParams(CBaseChainParams::REGTEST);
+    const BlockchainTestManager& testManager = BlockchainTestManager::GetInstance();
+
+    SidechainTxsCommitmentBuilder cmtObj;
+
+    CTransaction scCreationTx = txCreationUtils::createNewSidechainTxWith(CAmount(10), /*height*/10);
```

Comment:
Very low important and coming from previous commit; I think the comment `/*height*/` should be removed (the value `10` being `epochLength`); similar at lines 584 and 2824.

Reply:
Yes, corrected in e6db47d33